pkg/auth/iam/iam.go

```diff
@@ -320,27 +336,32 @@ func WithValidScope(scope string) FilterOption {
 }
 
 // WithMatchedSubdomain filters request to a subdomain to match it with namespace in user's token
+// Deprecated: subdomain validation will be done in the auth filter
 func WithMatchedSubdomain(excludedNamespaces []string) FilterOption {
```

Comment:
I can't follow "subdomain validation will be done in the auth filter" what is it mean? The validateSubdomainAgainstNamespace is still in the method.

Reply:
the `validateSubdomainAgainstNamespace` and `WithMatchedSubdomain` are essentially doing the same thing.
the difference is that previously we need to add the `WithMatchedSubdomain` filter option to all endpoints, now it is controlled by the init filter option `SubdomainValidationEnabled` and `SubdomainValidationExcludedNamespaces`

Reply:
ok, so my question:
1. do we need to change the services that used this method before?
2. if yes, can we directly remove this method? I assume this method is only used for foundations now.

Reply:
ok.